db/collection.go

```diff
@@ -228,25 +188,32 @@ func (db *db) createCollection(
 func (db *db) updateCollection(
 	ctx context.Context,
 	txn datastore.Txn,
-	existingDescriptionsByName map[string]client.CollectionDescription,
-	proposedDescriptionsByName map[string]client.CollectionDescription,
+	existingSchemaByName map[string]client.SchemaDescription,
+	proposedDescriptionsByName map[string]client.SchemaDescription,
 	desc client.CollectionDescription,
```

Comment:
todo: Passing this in no longer makes any sense, not does fetching it in patchSchema.  Drop the patchSchema call to `getCollectionsByName`, and fetch it on demand in here.

See also https://github.com/sourcenetwork/defradb/pull/1939#discussion_r1358762878

- [ ] Remove unwanted complexity RE collectionDescription param